sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1327,22 +1327,29 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
   override def visitComparison(ctx: ComparisonContext): Expression = withOrigin(ctx) {
     val left = expression(ctx.left)
     val right = expression(ctx.right)
+    val isUnsupportedComparison = (left, right) match {
```

Comment:
How about the case, `select * from t1 where (a > 3) = (b < 5) = (c > 3)` ?

Reply:
btw, have you check if we could update the ANTLR parser rules to prevent this case?